tests/phpunit/tests/rest-api/wpRestTemplateAutosavesController.php

```diff
@@ -0,0 +1,371 @@
+<?php
+/**
+ * Unit tests covering WP_REST_Template_Autosaves_Controller functionality.
+ *
+ * @package WordPress
+ * @subpackage REST API
+ *
+ * @group restapi
+ */
+class Tests_REST_wpRestTemplateAutosavesController extends WP_Test_REST_Controller_Testcase {
+
+	const TEST_THEME    = 'block-theme';
+	const TEMPLATE_NAME = 'my_template';
+
+	const PARENT_POST_TYPE = 'wp_template';
+
+	/**
+	 * Admin user ID.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @var int
+	 */
+	private static $admin_id;
+
+	/**
+	 * Template post.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @var WP_Post
+	 */
+	private static $template_post;
+
+	/**
+	 * Create fake data before our tests run.
+	 *
+	 * @param WP_UnitTest_Factory $factory Helper that lets us create fake data.
+	 */
+	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
+
+		self::$admin_id = $factory->user->create(
+			array(
+				'role' => 'administrator',
+			)
+		);
+		wp_set_current_user( self::$admin_id );
+
+		// Set up template post.
+		self::$template_post = $factory->post->create_and_get(
+			array(
+				'post_type'    => self::PARENT_POST_TYPE,
+				'post_name'    => self::TEMPLATE_NAME,
+				'post_title'   => 'My Template',
+				'post_content' => 'Content',
+				'post_excerpt' => 'Description of my template',
+				'tax_input'    => array(
+					'wp_theme' => array(
+						self::TEST_THEME,
+					),
+				),
+			)
+		);
+		wp_set_post_terms( self::$template_post->ID, self::TEST_THEME, 'wp_theme' );
+	}
+
+	/**
+	 * @covers WP_REST_Template_Autosaves_Controller::register_routes
+	 * @ticket 56922
+	 */
+	public function test_register_routes() {
+		$routes = rest_get_server()->get_routes();
+		$this->assertArrayHasKey(
+			'/wp/v2/templates/(?P<id>([^\/:<>\*\?"\|]+(?:\/[^\/:<>\*\?"\|]+)?)[\/\w%-]+)/autosaves',
+			$routes,
+			'Autosaves route does not exist.'
+		);
+		$this->assertArrayHasKey(
+			'/wp/v2/templates/(?P<parent>([^\/:<>\*\?"\|]+(?:\/[^\/:<>\*\?"\|]+)?)[\/\w%-]+)/autosaves/(?P<id>[\d]+)',
+			$routes,
+			'Single autosave based on the given ID route does not exist.'
+		);
```

Comment:
```suggestion
		$this->assertArrayHasKey(
			'/wp/v2/templates/(?P<id>([^\/:<>\*\?"\|]+(?:\/[^\/:<>\*\?"\|]+)?)[\/\w%-]+)/autosaves',
			$routes,
			'Template autosaves route does not exist.'
		);
		$this->assertArrayHasKey(
			'/wp/v2/templates/(?P<parent>([^\/:<>\*\?"\|]+(?:\/[^\/:<>\*\?"\|]+)?)[\/\w%-]+)/autosaves/(?P<id>[\d]+)',
			$routes,
			'Single template autosave based on the given ID route does not exist.'
		);
		$this->assertArrayHasKey(
			'/wp/v2/template-parts/(?P<id>([^\/:<>\*\?"\|]+(?:\/[^\/:<>\*\?"\|]+)?)[\/\w%-]+)/autosaves',
			$routes,
			'Template part autosaves route does not exist.'
		);
		$this->assertArrayHasKey(
			'/wp/v2/template-parts/(?P<parent>([^\/:<>\*\?"\|]+(?:\/[^\/:<>\*\?"\|]+)?)[\/\w%-]+)/autosaves/(?P<id>[\d]+)',
			$routes,
			'Single template part autosave based on the given ID route does not exist.'
		);
```

Reply:
@spacedmonkey I thought the template parts related tests would go into a separate test class. As far as I recall, we discussed that they could be added later. However, I don't mind adding them here. I'll commit the changes you proposed.